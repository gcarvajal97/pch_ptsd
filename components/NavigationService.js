import { NavigationActions } from 'react-navigation';
import { DrawerActions } from 'react-navigation-drawer';

// Props is apparently to be deprecated soon, I've researched the current best practice and created this "service"
// This service will allow us to make calls to move screens using the navigate function below
// I also added in an openDrawer function (so that it can be called from the header button on MainScreen.js)
let _navigator;

function setTopLevelNavigator(navigatorRef) {
  _navigator = navigatorRef;
}

function getTopLevelNavigator() {
  return _navigator;
}

// Can be used to navigate to other screens, but currently Drawer only references the stack navigator
function navigateDrawer(routeName, params) {
  _navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    })
  );
}

// Can be used to move into other screens defined in the stack navigator. 
function navigate(routeName, params) {
    _navigator.dispatch(
      NavigationActions.navigate({
        routeName,
        params,
      })
    );
  }

// Created this function to allow the openDrawer function to be called from the header in MainScreen.js
function openDrawer() {
    _navigator.dispatch(DrawerActions.openDrawer())
}

export default {
  navigate,
  navigateDrawer,
  setTopLevelNavigator,
  getTopLevelNavigator,
  openDrawer,
};