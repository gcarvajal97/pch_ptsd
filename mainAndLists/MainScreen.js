import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import React, { Component } from "react";
import { ScrollView, View } from "react-native";
import { Button } from "react-native-elements";
import FindHelpCard from '../components/FindHelpCard';
import LearnMoreCard from '../components/LearnMoreCard';
import NavigationService from '../components/NavigationService';
import QuizCard from '../components/QuizCard';
import { translate } from '../components/translateService';
import colors from '../colors';

// Set users locale for i18n translation throughout the application
i18n.locale = Localization.locale;

class MainScreen extends Component {

    static navigationOptions = () => ({
        headerTitle: translate('mainScreen.pediatricPTSD'),
        headerLeft: (
            <View style={{ margin: 5 }}>
                <Button
                onPress={()=>NavigationService.openDrawer()}
                title={translate('mainScreen.more')}
                titleStyle={{ fontStyle: null, fontWeight: null, fontFamily: "avenir-medium" }}
                buttonStyle={{ backgroundColor: colors.accent }}
                accessible
                accessibilityLabel={translate("mainScreen.more")}
                accessibilityHint={translate("glossary.backHint")}>
                </Button>
            </View>)
    })

    render() {
        return (
            <ScrollView>
                    <LearnMoreCard />
                    <QuizCard />
                    <FindHelpCard />
            </ScrollView>
        );
    }
}

export default MainScreen;


