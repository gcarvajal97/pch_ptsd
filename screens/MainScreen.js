import React, { Component } from "react";
import { View, Button, ScrollView } from "react-native";

import LearnMoreCard from '../components/LearnMoreCard';
import QuizCard from '../components/QuizCard';
import FindHelpCard from '../components/FindHelpCard';

import NavigationService from '../components/NavigationService';
    // I have learned that Props will become deprecated soon, this code has been changed to follow the
    // new best practice in React (to hopefully avoid features breaking in future updates).
    // NavigationService is a new file I created, allowing me to use the openDrawer function.
class MainScreen extends Component {

    static navigationOptions = () => ({
        headerTitle: 'Pediatric PTSD',
        headerLeft: (<View style={{ margin: 5 }}><Button title="Test"
            onPress={()=>NavigationService.openDrawer()}
            title='More'></Button></View>)
    })

    // Because we want the image, button and text to fit the width, the cards can only shrink so much.
    // I wrapped the Card's in a scrollview so that users on smaller screens can still access everything
    render() {
        return (
            <ScrollView>
                    <LearnMoreCard />
                    <QuizCard />
                    <FindHelpCard />
            </ScrollView>
        );
    }
}
export default MainScreen;