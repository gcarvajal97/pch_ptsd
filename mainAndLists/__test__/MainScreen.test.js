import React from 'react';
import MainScreen from '../MainScreen';
import NavigationService from '../../components/NavigationService';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
jest.mock('../../components/NavigationService');

describe("MainScreen", () => {
    it("matches Snapshot", () => {
        const snap = renderer.create(<MainScreen />).toJSON;
        expect(snap).toMatchSnapshot();
    });

    it('openDrawer() is called when header button is pressed', () => {
        const navProps = { navigate: () => {} };
        const navigationOptions = MainScreen.navigationOptions(navProps);
        navigationOptions.headerLeft.props.children.props.onPress();
        expect(NavigationService.openDrawer).toHaveBeenCalled();
    });
});