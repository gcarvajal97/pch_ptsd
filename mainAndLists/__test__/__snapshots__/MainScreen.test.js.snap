// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen matches Snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityLabel="Photo of a doctor holding a stethoscope with their arms folded accross their chest."
      accessible={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 0,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <View>
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "position": "relative",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                ]
              }
              testID="RNE__Image"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            />
            <View
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#bdbdbd",
                    "height": 150,
                    "justifyContent": "center",
                    "width": null,
                  }
                }
                testID="RNE__Image__placeholder"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "height": 150,
                    "width": null,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "rgba(0, 0, 0, 0.2)",
                    "bottom": 0,
                    "flex": 1,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 22.5,
                      "fontWeight": "800",
                      "marginBottom": 8,
                      "textAlign": "center",
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Learn More About Injury and Trauma
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Get up-to-date information and expert guidance to assist you and your injured child.
            </Text>
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessibilityHint="Navigates to the learn more screen"
                accessibilityLabel="LEARN MORE"
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="LearnMoreButton"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0077C8",
                      "borderColor": "#2089dc",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    LEARN MORE
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Photo of child covering her eyes with her hands"
      accessible={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 0,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <View>
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "position": "relative",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                ]
              }
              testID="RNE__Image"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            />
            <View
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#bdbdbd",
                    "height": 150,
                    "justifyContent": "center",
                    "width": null,
                  }
                }
                testID="RNE__Image__placeholder"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "height": 150,
                    "width": null,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "rgba(0, 0, 0, 0.2)",
                    "bottom": 0,
                    "flex": 1,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 22.5,
                      "fontWeight": "800",
                      "marginBottom": 8,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Rate Your Child's Reactions
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              An injury or accident can be a scary or stressful experience. Take a quick quiz to measure your child's traumatic stress.
            </Text>
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessibilityHint="Navigates to the quiz screen"
                accessibilityLabel="TAKE THE QUIZ"
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0077C8",
                      "borderColor": "#2089dc",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    TAKE THE QUIZ
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Photo of father reading a book to a baby on his lap."
      accessible={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 0,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <View>
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "position": "relative",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                ]
              }
              testID="RNE__Image"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            />
            <View
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#bdbdbd",
                    "height": 150,
                    "justifyContent": "center",
                    "width": null,
                  }
                }
                testID="RNE__Image__placeholder"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "height": 150,
                    "width": null,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "rgba(0, 0, 0, 0.2)",
                    "bottom": 0,
                    "flex": 1,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 22.5,
                      "fontWeight": "800",
                      "marginBottom": 8,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Help Your Child Recover
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Dealing with emotional reactions and injury care can be difficult. Find ways to help your child recover.
            </Text>
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessibilityHint="Navigates to the find help screen"
                accessibilityLabel="FIND HELP"
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="FindHelpButton"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0077C8",
                      "borderColor": "#2089dc",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    FIND HELP
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
